File: nodejs/tests/Ch.js
var Jstax = require('../jstax').Jstax;
module.exports =
{
	"read": function(test) {
		var input = new Jstax.Input.String('test');
		var t = new Jstax.Token.Ch('t');
		test.equal('t', t.read(input));
		test.done();
	},
	"matches": function(test) {
		var input = new Jstax.Input.String('test');
		var t = new Jstax.Token.Ch('t');
		var e = new Jstax.Token.Ch('e');
		test.equal(true, t.matches(input));
		test.equal(false, e.matches(input));
		test.done();
	}
};

File: compiler/src/main/java/org/hisrc/jstax/grammar/character/Ch.java
package org.hisrc.jstax.grammar.character;

import java.util.List;

import org.hisrc.jstax.grammar.gamma.Str;
import org.hisrc.jstax.io.ErrorHandler;
import org.hisrc.jstax.io.Input;
import org.hisrc.jstax.io.Result;

public interface Ch extends Str {

	public void read(Input input, Result result, ErrorHandler handler);

	public boolean startsInput(Input input);

	public boolean matches(char _char);

	public String toString();

	public <R> R accept(ChVisitor<R> visitor);

	public List<CharRange> getCharRanges();

	public boolean intersects(Ch that);

	public CharRanges minus(String name, Char that);
}
